src/components/WeekDayPicker/index.js

```diff
@@ -0,0 +1,144 @@
+import { useUniqueIdentifier } from '../../libs/hooks';
+import RenderIf from '../RenderIf';
+import WeekDayItems from './weekDayItems';
+import Label from '../Input/label';
+import StyledTextError from '../Input/styled/errorText';
+import { StyledFieldset, StyledHelpText } from './styled';
+
+/**
+ * A WeekDayPicker allows to select the days of the week
+ * @category Form
+ */
+function WeekDayPicker(props) {
+    const {
+        id,
+        name,
+        value,
+        label,
+        bottomHelpText,
+        availableDates,
+        disabled,
+        required,
+        readOnly,
+        multiple,
+        error,
+        onChange,
+        className,
+        style,
+    } = props;
+
+    const defaultFieldsetName = useUniqueIdentifier('week-day-items');
+    const fieldsetName = name || defaultFieldsetName;
+
+    const prepareReturnValue = (weekDayValue, isChecked) => {
+        if (multiple) {
+            if (isChecked && !value.includes(weekDayValue)) {
+                return sortWeekDays([...value, weekDayValue]);
+            }
+
+            return sortWeekDays(value.filter(day => day !== weekDayValue));
+        }
+
+        return weekDayValue;
+    };
+
+    const handleOnChange = e => {
+        const weekDayValue = e.currentTarget.value;
+        const isChecked = e.currentTarget.checked;
+
+        return onChange(prepareReturnValue(weekDayValue, isChecked));
+    };
+
+    return (
+        <StyledFieldset className={className} style={style} id={id}>
+            <Label label={label} hideLabel={!label} required={required} readOnly={readOnly} />
+            <WeekDayItems
+                name={fieldsetName}
+                value={value}
+                availableDates={availableDates}
+                disabled={disabled}
+                required={required}
+                readOnly={readOnly}
+                multiple={multiple}
+                error={error}
+                onChange={handleOnChange}
+            />
+            <RenderIf isTrue={!!bottomHelpText}>
+                <StyledHelpText>{bottomHelpText}</StyledHelpText>
+            </RenderIf>
+            <RenderIf isTrue={!!error}>
+                <StyledTextError>{error}</StyledTextError>
+            </RenderIf>
+        </StyledFieldset>
+    );
+}
+
+WeekDayPicker.propTypes = {
+    /** The id of the outer element. */
+    id: PropTypes.string,
+    /** An identifier for the group of radio items. */
+    name: PropTypes.string,
+    /** The value of WeekDayPicker, a list of days selected. */
+    value: PropTypes.oneOfType([
+        PropTypes.arrayOf((propValue, key, componentName, location, propFullName) => {
+            return valuePropValidation(propValue, key, componentName, location, propFullName);
+        }),
+        PropTypes.oneOf([
+            'monday',
+            'tuesday',
+            'wednesday',
+            'thursday',
+            'friday',
+            'saturday',
+            'sunday',
+            '',
```

Comment:
let's not included an empty string here

Reply:
If I don't include the empty string, it throws error when using WeekDayPicker for single day selection.

So, as I understand on this part, we will be allowing specific strings, but for the default value, If I don't add that '' empty default value, PropTypes throws the error ;(

If there is another possible solution I can apply, or maybe I need to some advice on this PropTypes definition. Maybe it is a little more difficult because we are limiting user to specific values (which is good practice on this case as I see).